### DISCUSSION TITLE
both distro provided qv2ray and AppImage can't bypass GFW while v2rayN with same remote server settings can

### DISCUSSION THREAD

**Discussion by zhirshya** [Qv2ray/qv2ray.github.io#125] [Q&A]
### Describe You Purpose
Hi, I tried both distribution provided qv2ray(2.6.3:5840) and latest AppImage(2.7.0+pre2) from your github release page on Linux Mint xfce 20.1 Ulyssa, without success, i.e. both versions of qv2ray can't circumvent GFW.

I want to access beyond GFW in Linux mint xfce 20.1 using qv2ray with the same remote server connection settings working with v2rayN on windows10.

I think the issues I encountered are somewhat different from
[2.7.0+pre seems not working with Linux distro shipped v2ray Qv2ray/Qv2ray#1294](url)
because qv2ray executable starts without obvious errors and v2raycore and asset directory settings are confirmed to be correct from qv2ray itself.

### Describe What Have You Tried
I followed steps in https://qv2ray.net/getting-started/ and checked the vcore executable and assets directory settings to be correct with the check settings buttion therein.
the log excerpt is:

```
2021/04/01 10:02:10 [Warning] [1705813862] github.com/v2fly/v2ray-core/v4/app/proxyman/outbound: failed to process outbound traffic > github.com/v2fly/v2ray-core/v4/proxy/vless/outbound: failed to find an available destination > github.com/v2fly/v2ray-core/v4/common/retry: [github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial WebSocket > github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial to (wss://198.41.208.0/ray):  > remote error: tls: handshake failure] > github.com/v2fly/v2ray-core/v4/common/retry: all retry attempts failed
2021/04/01 10:02:10 [Warning] [2641603753] github.com/v2fly/v2ray-core/v4/app/proxyman/outbound: failed to process outbound traffic > github.com/v2fly/v2ray-core/v4/proxy/vless/outbound: failed to find an available destination > github.com/v2fly/v2ray-core/v4/common/retry: [github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial WebSocket > github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial to (wss://198.41.208.0/ray):  > remote error: tls: handshake failure] > github.com/v2fly/v2ray-core/v4/common/retry: all retry attempts failed
2021/04/01 10:02:10 [Warning] [2641603753] github.com/v2fly/v2ray-core/v4/proxy/http: failed to read response from detectportal.firefox.com > io: read/write on closed pipe
2021/04/01 10:02:11 [Warning] [536152214] github.com/v2fly/v2ray-core/v4/app/proxyman/outbound: failed to process outbound traffic > github.com/v2fly/v2ray-core/v4/proxy/vless/outbound: failed to find an available destination > github.com/v2fly/v2ray-core/v4/common/retry: [github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial WebSocket > github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial to (wss://198.41.208.0/ray):  > remote error: tls: handshake failure] > github.com/v2fly/v2ray-core/v4/common/retry: all retry attempts failed
2021/04/01 10:02:12 [Warning] [2353383216] github.com/v2fly/v2ray-core/v4/app/proxyman/outbound: failed to process outbound traffic > github.com/v2fly/v2ray-core/v4/proxy/vless/outbound: failed to find an available destination > github.com/v2fly/v2ray-core/v4/common/retry: [github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial WebSocket > github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial to (wss://198.41.208.0/ray):  > remote error: tls: handshake failure] > github.com/v2fly/v2ray-core/v4/common/retry: all retry attempts failed
2021/04/01 10:02:12 [Warning] [1110819894] github.com/v2fly/v2ray-core/v4/app/proxyman/outbound: failed to process outbound traffic > github.com/v2fly/v2ray-core/v4/proxy/vless/outbound: failed to find an available destination > github.com/v2fly/v2ray-core/v4/common/retry: [github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial WebSocket > github.com/v2fly/v2ray-core/v4/transport/internet/websocket: failed to dial to (wss://198.41.208.0/ray):  > remote error: tls: handshake failure] > github.com/v2fly/v2ray-core/v4/common/retry: all retry attempts failed.
```

However, using same remote servers for v2rayN 4.12 and 4.13 on windows 10 can bypass GFW without problem.
Both Linux Mint xfce 20.1 Ulyssa and windows 10 aforementioned are guest VM inside Vmware.

So, I have 2 questions now:
1. What could be the problems for my settings of qv2ray on Linux Mint xfce 20.1 Ulyssa?
Why qv2ray using same server settings as v2rayN did can't be able to bypass GFW while v2rayN could?

2. I noticed the AppImage started with full path from terminal shows version 2.6.3:5840 in its About page in preference. That's the same version as distro provided qv2ray installed with apt.
It should show 2.7.0+pre2, shouldn't it?
Why the AppImage invoked with absolute path show version of distro provided one's?

### Any Other Thing
Maybe I missed something and ignorant of some other settings.

**Comment by DuckSoft** [comment]
Qv2ray itself does not provide any proxy feature. All that feature comes from upstream V2Ray Core.
Now that the V2Ray core launched successfully and you got that warning logs, the issues doesn't seem to be ours.
Sorely from the log it shows that your TLS handshake with remote server was interrupted by that GFW.

You may want to share with us your configuration details to help us repeat the problem, if possible.

**Reply by zhirshya** [reply]
thank you for your prompt response.
The connection details is:
`{
    "outbounds": [
        {
            "_QV2RAY_USE_GLOBAL_FORWARD_PROXY_": false,
            "mux": {
                "concurrency": 1,
                "enabled": null
            },
            "protocol": "vless",
            "sendThrough": "0.0.0.0",
            "settings": {
                "vnext": [
                    {
                        "address": "198.41.208.0",
                        "port": 443,
                        "users": [
                            {
                                "encryption": "none",
                                "id": "f82b5622-3d18-4318-b3c6-c6367baa2167"
                            }
                        ]
                    }
                ]
            },
            "streamSettings": {
                "network": "ws",
                "security": "tls",
                "sockopt": {
                    "mark": 0
                },
                "tcpSettings": {
                    "header": {
                        "request": {
                            "headers": {
                            },
                            "path": [
                            ]
                        },
                        "response": {
                            "headers": {
                            }
                        },
                        "type": "none"
                    }
                },
                "tlsSettings": {
                    "alpn": [
                    ]
                },
                "wsSettings": {
                    "path": "/ray"
                }
            },
            "tag": ""
        }
    ]
}
`

I can't think of any other settings I tried before.
Is there any settings should be done with CLI way not GUI way?
And, could you please tell me why AppImage started with full path from terminal shows version 2.6.3:5840 instead of 2.7.0+pre2?

**Reply by DuckSoft** [reply]
> I noticed the AppImage started with full path from terminal shows version 2.6.3:5840 in its About page in preference. That's the same version as distro provided qv2ray installed with apt. It should show 2.7.0+pre2, shouldn't it? Why the AppImage invoked with absolute path show version of distro provided one's?

Qv2ray runs in single-instanced mode. So maybe you haven't quit that old version, so the new version just waked the old one..

**Reply by zhirshya** [reply]
Thank you, could you please give some hints about why v2ray-core have TLS related problems in my mint xfce vm?
Is there any configurations have to be done for v2ray-core in addition to those settings in qv2ray's GUI mentioned in https://qv2ray.net/getting-started/ ?
